core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinderFactory.java

```diff
@@ -259,21 +254,28 @@ private <T> boolean verifyBinderTypeMatchesTarget(Binder<T, ?, ?> binderInstance
 	@SuppressWarnings("unchecked")
 	private <T> Binder<T, ConsumerProperties, ProducerProperties> getBinderInstance(String configurationName) {
 		if (!this.binderInstanceCache.containsKey(configurationName)) {
-			logger.info("Creating binder: " + configurationName);
-			BinderConfiguration binderConfiguration = this.binderConfigurations
-					.get(configurationName);
-			Assert.state(binderConfiguration != null,
-					"Unknown binder configuration: " + configurationName);
-			BinderType binderType = this.binderTypeRegistry
-					.get(binderConfiguration.getBinderType());
-			Assert.notNull(binderType, "Binder type "
-					+ binderConfiguration.getBinderType() + " is not defined");
-
+			this.logger.info("Creating binder: " + configurationName);
+			BinderConfiguration binderConfiguration = this.binderConfigurations.get(configurationName);
+			Assert.state(binderConfiguration != null, "Unknown binder configuration: " + configurationName);
+			BinderType binderType = this.binderTypeRegistry.get(binderConfiguration.getBinderType());
+			Assert.notNull(binderType, "Binder type " + binderConfiguration.getBinderType() + " is not defined");
 			Map<String, Object> binderProperties = new HashMap<>();
 			this.flatten(null, binderConfiguration.getProperties(), binderProperties);
 
-			ConfigurableApplicationContext binderProducingContext =
-					this.initializeBinderContextSimple(configurationName, binderProperties, binderType, binderConfiguration);
+			ConfigurableApplicationContext binderProducingContext;
+			if (this.binderChildContextInitializers.containsKey(configurationName)) {
+				this.logger.info("Using AOT pre-prepared initializer to construct binder child context for " + configurationName);
+				binderProducingContext = this.initializeBinderContextSimple(configurationName, binderProperties,
+						binderType, binderConfiguration, false);
+				GenericApplicationContext c = null;
+				this.binderChildContextInitializers.get(configurationName).initialize(binderProducingContext);
```

Comment:
In our face-to-face review with Chris we have realized that after AOT we have already all the bean definitions for this kind of child application context.
So, it looks like an overhead to still call that whole `initializeBinderContextSimple()` which with the provided configuration going through all the scanning and parsing stuff which in the end is replaces with this `initialize()` call result.

Reply:
Yep, good catch @artembilan . I will amend this.